src/components/AttachmentPicker/index.native.js

```diff
@@ -28,54 +35,77 @@ const documentPickerOptions = {
 };
 
 /**
- * Launch the DocumentPicker. Results are in same format as ImagePicker, so we can pass the repsonse to the
- * callback as is.
+ * Inform the users when they need to grant camera access and guide them to settings
+ */
+function showPermissionsAlert() {
+    Alert.alert(
+        'Expensify.cash does not have access to your camera. To enable access, tap Settings and turn on Camera.',
+        '',
+        [
+            {
+                text: 'Cancel',
+                style: 'cancel',
+            },
+            {
+                text: 'Settings',
+                onPress: () => Linking.openSettings(),
+            },
+        ],
+        {cancelable: false},
+    );
+}
+
+/**
+ * A generic handling when we don't know the exact reason for an error
+ *
+ * @param {String} message
+ */
+function showGeneralAlert(message) {
+    Alert.alert(
+        'An error is preventing us to use handle the attachment',
+        message, // Todo: maybe we don't want to show this as it's probably not human friendly
```

Comment:
I wouldn't include the `message` here, since - as you mentioned - it might not be user-friendly. 

I would also update the error message to be `Oops. An error occurred while selecting this attachment. Please try again.`

Reply:
Thanks, the intended text actually was "An error prevented us from selecting an attachment" because this can potentially happen anywhere from initially pressing "Camera" (or the other options) so the user might not have had the chance to pick something at all 

Also the first parameter of the alert is the alert modal title, having something lengthy there might get truncated on Android like here:
![image](https://user-images.githubusercontent.com/12156624/113135685-121be880-922b-11eb-9d7a-53d2003f0d32.png)
- this is already mentioned and handled separately

I'll address this and let you decide if it's good

Reply:
I've update the error to appear like so:

#### Android:
![image](https://user-images.githubusercontent.com/12156624/113142366-7642aa80-9233-11eb-9f60-8e9542aa276b.png)

#### iOS (system dark theme):
![image](https://user-images.githubusercontent.com/12156624/113142589-b99d1900-9233-11eb-943e-e451b9c5dc9d.png)

Reply:
Resolving this as a new thread provided the message

Reply:
That looks good!